com.ibm.streamsx.metrics/impl/java/src/com/ibm/streamsx/metrics/CustomMetricsSink.java

```diff
@@ -0,0 +1,168 @@
+/* Generated by Streams Studio: November 29, 2016 at 3:22:39 PM EST */
+package com.ibm.streamsx.metrics;
+
+
+import java.net.InetAddress;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.log4j.Logger;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.client.transport.TransportClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.transport.InetSocketTransportAddress;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.transport.client.PreBuiltTransportClient;
+import org.influxdb.InfluxDB;
+import org.influxdb.InfluxDBFactory;
+import org.influxdb.InfluxDB.ConsistencyLevel;
+import org.influxdb.dto.BatchPoints;
+import org.influxdb.dto.Point;
+
+import com.ibm.streams.operator.AbstractOperator;
+import com.ibm.streams.operator.OperatorContext;
+import com.ibm.streams.operator.StreamingData.Punctuation;
+import com.ibm.streams.operator.StreamingInput;
+import com.ibm.streams.operator.Tuple;
+import com.ibm.streams.operator.model.InputPortSet;
+import com.ibm.streams.operator.model.InputPortSet.WindowMode;
+import com.ibm.streams.operator.model.InputPortSet.WindowPunctuationInputMode;
+import com.ibm.streams.operator.model.InputPorts;
+import com.ibm.streams.operator.model.Libraries;
+import com.ibm.streams.operator.model.Parameter;
+import com.ibm.streams.operator.model.PrimitiveOperator;
+
+@PrimitiveOperator(name="CustomMetricsSink", namespace="com.ibm.streamsx.metrics",
+description="Java Operator CustomMetricsSink")
```

Comment:
> simply takes tuple attribute names as a parameter and outputs it to the database

Seems like it has nothing to do with metrics, in the way Streams uses the term 'metric'.

Is it just a Sink for Grafana and Kibana?

Once there's a full description of what it does it will be easier to name it, and determine if it belongs in this toolkit.